staging/src/k8s.io/component-base/config/serializer/encode.go

```diff
@@ -0,0 +1,102 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package serializer
+
+import (
+	"fmt"
+	"reflect"
+
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/runtime/serializer"
+	"k8s.io/apimachinery/pkg/runtime/serializer/json"
+)
+
+// EncodingFormat is a high-level, typed constant configuring how an object should be encoded.
+type EncodingFormat string
+
+const (
+	// ContentTypeYAML encodes a config file in the YAML format.
+	ContentTypeYAML EncodingFormat = "application/yaml"
+	// ContentTypeJSON encodes a config file in compact JSON format.
+	ContentTypeJSON EncodingFormat = "application/json"
+)
+
+// NewConfigSerializer creates a new implementation of the ConfigSerializer interface.
+func NewConfigSerializer(scheme *runtime.Scheme, codecs *serializer.CodecFactory) ConfigSerializer {
```

Comment:
what are the requirements for this to work? The scheme needs the types and the codec must be a matching one? I would even expect an higher-level interface for that where this plumbing is automatically done, i.e. only type `Install` funcs are given and it just works.

Reply:
can you sketch in a comment what higher-level interface you'd want? Like... how would this be used in scheme.go?

I was thinking like this:
```
var (
   Scheme = runtime.NewScheme()
   Codecs = serializer.NewCodecFactory(scheme)
   Serializer = configserializer.NewConfigSerializer(scheme, codecs)
   Encode = Serializer.Encode
   Decode = Serializer.Decode
)
```